src/test/java/ca/uhn/fhir/jpa/starter/OperationCollectDataTest.java

```diff
@@ -125,33 +280,58 @@ void beforeEach() {
 	}
 
 	private Parameters fetchParameter(String theUrl) throws IOException {
-		Parameters parameters;
 		HttpGet get = new HttpGet(theUrl);
 
 		String auth = USER_NAME + ":" + USER_PASSWORD;
 		byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
 		String authHeader = "Basic " + new String(encodedAuth);
 		get.addHeader(HttpHeaders.AUTHORIZATION, authHeader);
-		get.addHeader(HttpHeaders.ACCEPT, "application/fhir+json");
 
 		try (CloseableHttpResponse resp = ourHttpClient.execute(get)) {
-			parameters = ourCtx.newJsonParser().parseResource(Parameters.class,
-					EntityUtils.toString(resp.getEntity(), Charsets.UTF_8));
+			IParser parser;
+			HttpEntity httpEntity = resp.getEntity();
+			Header contentType = httpEntity.getContentType();
+			String mimeType = contentType.getValue();
+
+			MediaType responseMediaType = MediaType.parseMediaType(mimeType);
+			if (FHIR_JSON_MEDIA_TYPE.isCompatibleWith(responseMediaType) || MediaType.APPLICATION_JSON.isCompatibleWith(responseMediaType)) {
+				parser = ourCtx.newJsonParser();
+			} else if (FHIR_XML_MEDIA_TYPE.isCompatibleWith(responseMediaType) || MediaType.APPLICATION_XML.isCompatibleWith(responseMediaType) || MediaType.TEXT_XML.isCompatibleWith(responseMediaType)) {
+				parser = ourCtx.newXmlParser();
+			}
+			 else {
+				throw new AssertionError("Cannot handle response of type " + responseMediaType.toString());
+			}
+	
+			return parser.parseResource(Parameters.class,
+				  EntityUtils.toString(resp.getEntity(), Charsets.ISO_8859_1));
```

Comment:
We should be using `Charsets.UTF_8` as this is the default encoding for both XML and JSON files. Roughly speaking UTF-8 is a superset of ISO 8859-1.

Reply:
I have fixed it